addons/binding/org.openhab.binding.nibeheatpump/src/main/java/org/openhab/binding/nibeheatpump/handler/NibeHeatPumpHandler.java

```diff
@@ -0,0 +1,544 @@
+                } catch (ExecutionException e) {
+                    logger.error("Message sending to heat pump failed, exception {}", e.getMessage());
+                } finally {
+                    writeResult = null;
+                }
+
+                // Clear cache value to refresh coil data from the pump.
+                // We might not know if write message have succeed or not, so let's always refresh it.
+                logger.debug("Clearing cache value for channel '{}' to refresh channel data", channelUID);
+                clearCache(coilAddress);
+
+            } else {
+                logger.error("Command to channel '{}' rejected, because item is read only parameter", channelUID);
+            }
+        } else {
+            logger.warn("No connection to heat pump");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_MISSING_ERROR);
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.debug("channelLinked: {}", channelUID);
+
+        // Add channel to polling loop
+        int coilAddress = parseCoilAddressFromChannelUID(channelUID);
+        synchronized (itemsToPoll) {
+            itemsToPoll.add(coilAddress);
+        }
+        clearCache(coilAddress);
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+        logger.debug("channelUnlinked: {}", channelUID);
+
+        // remove channel from polling loop
+        int coilAddress = parseCoilAddressFromChannelUID(channelUID);
+        synchronized (itemsToPoll) {
+            itemsToPoll.remove(coilAddress);
+        }
+    }
+
+    private int parseCoilAddressFromChannelUID(ChannelUID channelUID) {
+        if (channelUID.getId().contains("#")) {
+            String[] parts = channelUID.getId().split("#");
+            return Integer.parseInt(parts[parts.length - 1]);
+        } else {
+            return Integer.parseInt(channelUID.getId());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initialized Nibe Heat Pump device handler for {}", getThing().getUID());
+        configuration = getConfigAs(NibeHeatPumpConfiguration.class);
+        logger.info("Using configuration: {}", configuration.toString());
+
+        try {
+            parseWriteEnabledItems();
+            connector = ConnectorFactory.getConnector(thing.getThingTypeUID());
+        } catch (IllegalArgumentException | NibeHeatPumpException e) {
+            String description = String.format("Illegal configuration, %s", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, description);
+            return;
+        }
+
+        clearCache();
+
+        if (connectorTask == null || connectorTask.isCancelled()) {
+            connectorTask = scheduler.scheduleAtFixedRate(new Runnable() {
```

Comment:
could you also use `scheduleWithFixedDelay` instead?

Reply:
Done